// @shopify/react-network/dist/src/hooks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
var accept_language_parser_1 = require("accept-language-parser");
var network_1 = require("@shopify/network");
var react_effect_1 = require("@shopify/react-effect");
var context_1 = require("./context");
function useNetworkEffect(perform) {
    var network = react_1.useContext(context_1.NetworkContext);
    react_effect_1.useServerEffect(function () {
        if (network != null) {
            return perform(network);
        }
    }, network ? network.effect : undefined);
}
exports.useNetworkEffect = useNetworkEffect;
function useCspDirective(directive, source) {
    useNetworkEffect(function (network) { return network.addCspDirective(directive, source); });
}
exports.useCspDirective = useCspDirective;
function useRequestHeader(header) {
    var network = react_1.useContext(context_1.NetworkContext);
    return network ? network.getHeader(header) : undefined;
}
exports.useRequestHeader = useRequestHeader;
function useHeader(header, value) {
    useNetworkEffect(function (network) { return network.setHeader(header, value); });
}
exports.useHeader = useHeader;
function useNetworkManager() {
    return react_1.useContext(context_1.NetworkContext);
}
exports.useNetworkManager = useNetworkManager;
function useStatus(code) {
    useNetworkEffect(function (network) { return network.addStatusCode(code); });
}
exports.useStatus = useStatus;
function useRedirect(url, status) {
    useNetworkEffect(function (network) { return network.redirectTo(url, status); });
}
exports.useRedirect = useRedirect;
function useAcceptLanguage(fallback) {
    if (fallback === void 0) { fallback = { code: 'en', quality: 1.0 }; }
    var acceptsLanguages = useRequestHeader(network_1.Header.AcceptLanguage);
    var locales = acceptsLanguages ? accept_language_parser_1.parse(acceptsLanguages) : [fallback];
    return locales;
}
exports.useAcceptLanguage = useAcceptLanguage;
